### PR TITLE
Auto formatting

### PR DESCRIPTION
**Pull request by paulgoetze**
There is no functional changes in this PR, just a bit of house-keeping.

 It adds the configuration and applies mix’s built-in auto-formatting and adds a log file to .gitignore that is created when running tests locally.

### CODE REVIEW

File: lib/rooms/room.ex [hunk 1]
```diff
@@ -35,51 +35,56 @@ defmodule SignalTower.Room do
...
         {:reply, :error, state}
     end
   end
 
   defp get_next_id(members) do
     case map_size(members) do
-      0 -> "0"
+      0 ->
+        "0"
+
```
Comment (farao) [inline lib/rooms/room.ex:72]: that's a strange empty line, does the formatter add it?

File: lib/rooms/room.ex [hunk 2]
```diff
@@ -95,24 +100,27 @@ defmodule SignalTower.Room do
...
     |> case do
-      {id,_} ->
+      {id, _} ->
         case leave(id, state) do
-          {:ok, state}   -> {:noreply, state}
-          {:error,state} -> {:noreply, state}
-          {:stop,state}  -> {:stop, :normal, state}
+          {:ok, state} -> {:noreply, state}
+          {:error, state} -> {:noreply, state}
+          {:stop, state} -> {:stop, :normal, state}
```
Comment (farao) [inline lib/rooms/room.ex:111]: that's the disadvantage of the formatter :D but it's ok

File: lib/rooms/stats.ex [hunk 3]
```diff
@@ -28,25 +31,34 @@ defmodule SignalTower.Stats do
...
+          Enum.join(
+            [
+              s.create_time |> DateTime.to_string(),
+              s.peak_time |> DateTime.to_string(),
+              DateTime.utc_now() |> DateTime.to_string(),
+              s.peak_user_count
+            ],
+            ","
+          )
+
```
Comment (farao) [inline lib/rooms/stats.ex:61]: hm, was before a bit easier to read, wasn't it?

File: lib/signal_tower.ex [hunk 4]
```diff
@@ -15,22 +14,27 @@ defmodule SignalTower do
...
 
-    dispatch = :cowboy_router.compile([
-      {"localhost", [{"/metrics", SignalTower.PrometheusHTTPHandler, []}]},
-      {:_,          [{"/", SignalTower.WebsocketHandler, []}]}
-    ])
+    dispatch =
+      :cowboy_router.compile([
+        {"localhost", [{"/metrics", SignalTower.PrometheusHTTPHandler, []}]},
+        {:_, [{"/", SignalTower.WebsocketHandler, []}]}
+      ])
```
Comment (farao) [inline lib/signal_tower.ex:21]: same here

File: lib/users/msg_integrity.ex [hunk 5]
```diff
@@ -14,29 +14,42 @@ defmodule SignalTower.MsgIntegrity do
...
-    msg = cond do
-      msg["event"] == "join_room" && !msg["status"] ->
-        Map.put msg, "status", %{}
-      true -> msg
-    end
+    msg =
+      cond do
+        msg["event"] == "join_room" && !msg["status"] ->
+          Map.put(msg, "status", %{})
+
```
Comment (farao) [inline lib/users/msg_integrity.ex:26]: also here: strange empty line, right?

Comment (farao) [reply lib/users/msg_integrity.ex:26]: I stop from here pointing this out :D is this formatter related?

File: test/room_test.exs [hunk 6]
```diff
@@ -54,55 +67,68 @@ defmodule RoomTest do
...
+                   1000
   end
 
   # session process dies
   test "user leaves room when his session dies" do
     room_pid = create_room("r-room6")
     join_room(self(), room_pid)
 
-    spawn_link fn ->
+    spawn_link(fn ->
```
Comment (farao) [inline test/room_test.exs:111]: I know, this is better, but it was so much prettier before :-)

**Comment by farao** [comment]
Generally, a huge improvement. As pointed out, I'm a bit sad about not aligning things (e.g. arrows) anymore and the strange empty lines between e.g. cond cases, but if it makes it easier to have a coherent format then I can accept it

**Comment by paulgoetze** [comment]
@farao I see where you are coming from in terms of formatting, but I figure that's the reason for why a formatter was added to Elixir in the first place - to not have to argue about it anymore ;) As far as I could see there is no options or customizations of the formatting. Maybe also if it looks weird it can be done in a better/different way...?

I would still stick to the official formatting just for consistency reasons. And it's probably a huge time saver.

**Comment by farao** [comment]
True! Thanks :-)

**Pull request merged**